near/receiver/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+    }
+
+    /// Return the deposit required to update a price feed.
+    pub fn get_update_fee_estimate(&self) -> u64 {
+        let byte_cost = env::storage_byte_cost();
+        let data_cost = byte_cost * std::mem::size_of::<PriceFeed>() as u128;
+        4u64 * u64::try_from(data_cost).unwrap() + self.update_fee
+    }
+
+    #[payable]
+    #[private]
+    #[handle_result]
+    pub fn verify_vaa_callback(
+        &mut self,
+        account_id: AccountId,
+        vaa: String,
+        #[callback_result] _result: Result<u32, near_sdk::PromiseError>,
+    ) -> Result<(), Error> {
+        if !is_promise_success() {
+            return Err(Error::VaaVerificationFailed);
+        }
+
+        // Get Storage Usage before execution, subtracting the fee from the deposit has the effect
+        // forces the caller to add the required fee to the deposit.
+        let storage = env::storage_usage()
+            .checked_sub(self.update_fee)
+            .ok_or(Error::InsufficientDeposit)?;
+
+        // Deserialize VAA, note that we already deserialized and verified the VAA in `process_vaa`
+        // at this point so we only care about the `rest` component which contains bytes we can
+        // deserialize into an Action.
+        let vaa = hex::decode(&vaa).unwrap();
+        let (_, rest): (wormhole::Vaa<()>, _) =
+            serde_wormhole::from_slice_with_payload(&vaa).unwrap();
+
+        // Attempt to deserialize the Batch of Price Attestations.
+        let bytes = &mut Cursor::new(rest);
+        let batch = BatchPriceAttestation::deserialize(bytes).unwrap();
+
+        // Verify the PriceAttestation's are new enough, and if so, store them.
+        let mut count_updates = 0;
+        for price_attestation in &batch.price_attestations {
+            if self.update_price_feed_if_new(PriceFeed::from(price_attestation)) {
+                count_updates += 1;
+            }
+        }
+
+        log!(
+            r#"
+            {{
+                "standard": "pyth",
+                "version":  "1.0",
+                "event":    "BatchAttest",
+                "data":     {{
+                    "count": {},
+                    "diffs": {},
+                    "costs": {},
+                }}
+            }}
+        "#,
+            count_updates,
+            env::storage_usage() - storage,
+            env::storage_byte_cost() * (env::storage_usage() - storage) as u128,
+        );
+
+        // Refund storage difference to `account_id` after storage execution.
+        self.refund_storage_usage(
+            account_id,
+            storage,
+            env::storage_usage(),
+            env::attached_deposit(),
+        )
+    }
+
+    /// Read the list of accepted `Source` chains for a price attestation.
+    pub fn get_sources(&self) -> Vec<Source> {
+        self.sources.iter().collect()
+    }
+
+    /// Get the current staleness threshold.
+    pub fn get_stale_threshold(&self) -> u64 {
+        self.stale_threshold
```

Comment:
suggest naming this variable something like `stale_threshold_secs` so the unit is clear

Reply:
maybe change u64 to `Duration` from near. I'm confused why near has a duration, is the rust Duration not accessible in Near? Also now that `Duration` is near is nanoseconds we should make sure to convert it properly.